docarray/typing/url/any_url.py

```diff
@@ -38,6 +52,29 @@ def _to_node_protobuf(self) -> 'NodeProto':
 
         return NodeProto(text=str(self), type=self._proto_type_name)
 
+    @classmethod
+    def is_extension_allowed(cls, value: Any) -> bool:
+        """
+        Check if the file extension of the URL is allowed for this class.
+        First, it guesses the mime type of the file. If it fails to detect the
+        mime type, it then checks the extra file extension.
+
+        :param value: The URL or file path.
+        :return: True if the extension is allowed, False otherwise
+        """
+        if cls is AnyUrl:
+            return True
```

Comment:
I do not understand this code, will this not always return True? what ks trying to do here?

Reply:
No this is a direct comparison (same as ==)
issubclass would return True always